presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -517,11 +528,47 @@ private ConnectorTableMetadata doGetTableMetadata(SchemaTableName tableName)
             properties.put(TEXTFILE_SKIP_FOOTER_LINE_COUNT, Integer.valueOf(textSkipFooterCount));
         }
 
+        // CSV specific property
+        getCsvSerdeProperty(table.get(), CSV_SEPARATOR_KEY)
+                .ifPresent(csvSeparator -> properties.put(CSV_SEPARATOR, csvSeparator));
+        getCsvSerdeProperty(table.get(), CSV_QUOTE_KEY)
+                .ifPresent(csvQuote -> properties.put(CSV_QUOTE, csvQuote));
+        getCsvSerdeProperty(table.get(), CSV_ESCAPE_KEY)
+                .ifPresent(csvEscape -> properties.put(CSV_ESCAPE, csvEscape));
+
         Optional<String> comment = Optional.ofNullable(table.get().getParameters().get(TABLE_COMMENT));
 
         return new ConnectorTableMetadata(tableName, columns.build(), properties.build(), comment);
     }
 
+    private static Optional<String> getCsvSerdeProperty(Table table, String key)
+    {
+        return getSerdeProperty(table, key).map(csvSerdeProperty -> {
+            if (csvSerdeProperty.length() > 1) {
+                throw new PrestoException(HIVE_INVALID_METADATA, "Only single character can be set for property: " + key);
+            }
+            return csvSerdeProperty;
+        });
+    }
+
+    private static Optional<String> getSerdeProperty(Table table, String key)
+    {
+        String serdePropertyValue = table.getStorage().getSerdeParameters().get(key);
+        String tablePropertyValue = table.getParameters().get(key);
+        if (serdePropertyValue == null) {
+            return Optional.ofNullable(tablePropertyValue);
+        }
+        if (tablePropertyValue == null) {
+            return Optional.of(serdePropertyValue);
+        }
+        if (!tablePropertyValue.equals(serdePropertyValue)) {
+            throw new PrestoException(
+                    HIVE_INVALID_METADATA,
+                    format("Different values for '%s' set in serde properties '%s' and table properties '%s'", key, serdePropertyValue, tablePropertyValue));
```

Comment:
> What's hive's behavior when property declared twice, differently?

https://gist.github.com/kokosing/ffbc55bd3a8aa5bf8ad56ea89a7bffd8

table properties win

Reply:
Please keep this knowledge in the code comment.
I am OK with failing when they differ (even if hive doesn't)

Reply:
It could be that hive may work differently for different serde (or different input format). Here this method is generic. I can add a comment explaining how it works for CSV only.